lib/active_storage/service/mirror_service.rb

```diff
@@ -26,9 +27,9 @@ def primary_service
     end
 
     def perform_across_services(method, *args)
-      # FIXME: Convert to be threaded
-      services.collect do |service|
-        service.public_send method, *args
+      promises = services.collect do |service|
+        Concurrent::Promise.execute { service.public_send method, *args }
       end
+      Concurrent::Promise.zip(*promises).value
```

Comment:
Makes one wish for a simple `services.pmap do |service|` /cc @jdantonio 😊

Ref. https://github.com/ruby-concurrency/concurrent-ruby/issues/493#issuecomment-170250964